core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.kt

```diff
@@ -0,0 +1,290 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.webserver
+
+import android.app.ProgressDialog
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.Bundle
+import android.os.IBinder
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.widget.Toolbar
+import kotlinx.android.synthetic.main.activity_zim_host.recyclerViewZimHost
+import kotlinx.android.synthetic.main.activity_zim_host.serverTextView
+import kotlinx.android.synthetic.main.activity_zim_host.startServerButton
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.utils.AlertDialogShower
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import org.kiwix.kiwixmobile.core.utils.ServerUtils
+import org.kiwix.kiwixmobile.core.wifi_hotspot.HotspotService
+import org.kiwix.kiwixmobile.core.wifi_hotspot.HotspotService.ACTION_CHECK_IP_ADDRESS
+import org.kiwix.kiwixmobile.core.wifi_hotspot.HotspotService.ACTION_START_SERVER
+import org.kiwix.kiwixmobile.core.wifi_hotspot.HotspotService.ACTION_STOP_SERVER
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.SelectionMode
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BookOnDiskDelegate
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskAdapter
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem.BookOnDisk
+import java.util.ArrayList
+import javax.inject.Inject
+
+class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
+  @Inject
+  internal lateinit var presenter: ZimHostContract.Presenter
+  @Inject
+  internal lateinit var alertDialogShower: AlertDialogShower
+  private lateinit var booksAdapter: BooksOnDiskAdapter
+  private lateinit var bookDelegate: BookOnDiskDelegate.BookDelegate
+  private var hotspotService: HotspotService? = null
+  private var ip: String? = null
+  private var serviceConnection: ServiceConnection? = null
+  private var progressDialog: ProgressDialog? = null
+  private val tag = "ZimHostActivity"
+  private val ipStateKey = "ip_state_key"
+
+  private val selectedBooksPath: ArrayList<String>
+    get() {
+      val selectedBooksPath = ArrayList<String>()
```

Comment:
We should kotlin this up, whenever you declare a variable and iterate a list to populate that variable you most likely don't need to do that.

```
return booksAdapter.items.filter { it.isSelected }.filterIsInstance<BookOnDisk>().map { it.file.absolutePath }
```

Reply:
https://github.com/kiwix/kiwix-android/pull/1569/commits/33fa1ad1170d1500793c1f018271fb79bbca721e